### PR TITLE
feat(storage): sample gap (1/2 object part)

### PR DESCRIPTION
**Pull request by AlisskaPie**
Added samples with following region tags:
- [x] storage_generate_encryption_key 
- [x] storage_enable_versioning
- [x] storage_disable_versioning
- [x] storage_copy_file_archived_generation
- [x] storage_delete_file_archived_generation
- [x] storage_get_service_account

First part(objects) from [this issue](https://github.com/GoogleCloudPlatform/golang-samples/issues/1707)

### CODE REVIEW

**Comment by AlisskaPie** [comment]
@tritone @tbpg PTAL

File: storage/objects/generate_encryption_key.go
Comment (tritone) [inline storage/objects/generate_encryption_key.go:26]: I notice that the C++ sample here has a lot more information about encryption. Do we want to include that in the other samples (or pull up into the docs)? @frankyn  https://cloud.google.com/storage/docs/encryption/using-customer-supplied-keys#generating_your_own_encryption_key

**Review by tritone — COMMENTED** [review]
Again, let's wait for @JesseLovelace to review before merging.

File: storage/objects/generate_encryption_key.go
Comment (JesseLovelace) [reply storage/objects/generate_encryption_key.go:26]: The C++ sample seems a little excessive to me, but some other samples have a little piece of clarification text that I think should be included here:
```
    Generates a 256 bit (32 byte) AES encryption key and prints the
    base64 representation.

    This is included for demonstration purposes. You should generate your own
    key. Please remember that encryption keys should be handled with a
    comprehensive security policy.
```

**Review by JesseLovelace — CHANGES_REQUESTED** [review]
Only note is about the comment in key generation sample, other than that looks good

File: storage/objects/generate_encryption_key.go
Comment (AlisskaPie) [reply storage/objects/generate_encryption_key.go:27]: Done

File: storage/service_account/hmac/hmac_test.go
Comment (AlisskaPie) [reply storage/service_account/hmac/hmac_test.go]: Done

File: storage/objects/copy_file_archived_generation.go
Comment (tritone) [inline storage/objects/copy_file_archived_generation.go:28]: Change `srcBucket` -> `bucket` (since there's only one now).

**Review by tritone — COMMENTED** [review]
Couple nits, otherwise looks good!

File: storage/service_account/service_account.go
Comment (tritone) [inline storage/service_account/service_account.go:34]: Keep a context timeout, but set to 10s instead of 1 min.

File: storage/objects/copy_file_archived_generation.go
Comment (AlisskaPie) [reply storage/objects/copy_file_archived_generation.go]: Done

File: storage/service_account/service_account.go
Comment (AlisskaPie) [reply storage/service_account/service_account.go:35]: Done

**Review by tritone — APPROVED** [review]

**Review by tbpg — APPROVED** [review]

**Pull request merged**
